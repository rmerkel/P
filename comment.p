{ "main" starts here... }
var n, f;	{ var z; parser doesn't see 'z' }
begin
   n = 0;
   f = 1;
   {	calculate factor (n)
		comment continued on this line... }
   while n < 10 do begin
      n = n + 1;
      f = f * n
   end
end.

{	unterminated comment, but we don't care as it follows the period!
