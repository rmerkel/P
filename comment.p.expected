# comment.p, 2: { "main" starts here... }
# comment.p, 3: var n, f;	{ var z; parser doesn't see 'z' }
    0: call 0, 2
    1: halt
# comment.p, 4: begin
    2: enter 2
# comment.p, 5:    n = 0;
    3: pushConst 0
    4: pushVar 0, 4
    5: assign
# comment.p, 6:    f = 1;
    6: pushConst 1
    7: pushVar 0, 5
    8: assign
# comment.p, 7:    {	calculate factor (n)
# comment.p, 8: 		comment continued on this line... }
# comment.p, 9:    while n < 10 do begin
    9: pushVar 0, 4
   10: eval
   11: pushConst 10
   12: lt
   13: jneq 28
# comment.p, 10:       n = n + 1;
   14: pushVar 0, 4
   15: eval
   16: pushConst 1
   17: addi
   18: pushVar 0, 4
   19: assign
# comment.p, 11:       f = f * n
   20: pushVar 0, 5
   21: eval
# comment.p, 12:    end
   22: pushVar 0, 4
   23: eval
   24: muli
   25: pushVar 0, 5
   26: assign
# comment.p, 13: end.
   27: jump 9
   28: ret
#comment.p, 13: 
#comment.p, 14: {	unterminated comment, but we don't care as it follows the period!

        8:          0
        9:          1
        8:          1
        9:          1
        8:          2
        9:          2
        8:          3
        9:          6
        8:          4
        9:         24
        8:          5
        9:        120
        8:          6
        9:        720
        8:          7
        9:       5040
        8:          8
        9:      40320
        8:          9
        9:     362880
        8:         10
        9:    3628800
